packages/redux/src/settings/reducer/setting.ts

```diff
@@ -0,0 +1,69 @@
+import * as actionTypes from '../actionTypes.js';
+import { type AnyAction, combineReducers, type Reducer } from 'redux';
+import type { BlackoutError } from '@farfetch/blackout-client';
+import type { SettingState } from '../types/index.js';
+
+export const INITIAL_STATE: SettingState = {
+  error: null,
+  isLoading: false,
+  result: null,
+};
+
+const error = (
+  state: BlackoutError | null = INITIAL_STATE.error,
+  action: AnyAction,
+) => {
+  switch (action.type) {
+    case actionTypes.FETCH_SETTING_REQUEST:
+      return INITIAL_STATE.error;
+    case actionTypes.FETCH_SETTING_FAILURE:
+      return action.payload.error;
+    default:
+      return state;
+  }
+};
+
+const isLoading = (state = INITIAL_STATE.isLoading, action: AnyAction) => {
+  switch (action.type) {
+    case actionTypes.FETCH_SETTING_REQUEST:
+      return true;
+    case actionTypes.FETCH_SETTING_SUCCESS:
+    case actionTypes.FETCH_SETTING_FAILURE:
+      return INITIAL_STATE.isLoading;
+    default:
+      return state;
+  }
+};
+
+const result = (state = INITIAL_STATE.result, action: AnyAction) => {
+  switch (action.type) {
+    case actionTypes.FETCH_SETTING_SUCCESS:
```

Comment:
Should we clear the result when there is an error?

Reply:
No, if we can accumulate the settings on the redux state, we should maintain already fetched information.